### ISSUE TITLE
Generate OpenAPI `example` from annotation

### ISSUE THREAD

**Issue by cari-lynn** [k14s/ytt#557]
## Why
A configuration author who needs to show a valid reasonable value for a key, that does not belong in a `default` value, may put it in the `example` key.

To quote the [spec](https://swagger.io/docs/specification/adding-examples/):
> Note that schemas and properties support single example but not multiple examples.

OpenAPI 3.0 only allows singular `example` key in the `components.schemas` section. (It does allow `examples` elsewhere in a document though). This key can be any type (depending on where you read).

### 🍏  Scenario 1: `@schema/examples` creates a single `example` key with a description

**Given** the following schema exists
```yaml
#! schema.yml
#@data/values-schema
---
#@schema/examples ("Exposes the Service on each Node's IP at a static port", "NodePort")
service_type: ClusterIP
```
Note: The argument is a Tuple, with the first string being a description of the example, and the second being the example value.

**When** I execute `ytt -f schema.yml --data-values-schema-inspect -o openapi-v3`
**Then** I should see an `example` and a `x-example-description` key added
```yaml
openapi: 3.0.0
info:
  version: 0.1.0
  title: Openapi schema generated from ytt Data Values Schema
paths: {}
components: 
  schemas:
    dataValues:
      type: object
      additionalProperties: false
      properties:
        service_type:
          type: string
          default: ClusterIP
          x-example-description: Exposes the Service on each Node's IP at a static port
          example: NodePort 
```
NOTE: The `x-example-description` is a [specification extension](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.3.md#specification-extensions)

### 🍏  Scenario 2: `@schema/examples` creates a single `example` key

**Given** the following schema exists
```yaml
#! schema.yml
#@data/values-schema
---
#@schema/examples ( "NodePort",)
service_type: ClusterIP
```
Note: The argument is a Tuple, with the first string being a description of the example, and the second being the example value.

**When** I execute `ytt -f schema.yml --data-values-schema-inspect -o openapi-v3`
**Then** I should see NO `x-example-description` key
```yaml
openapi: 3.0.0
info:
  version: 0.1.0
  title: Openapi schema generated from ytt Data Values Schema
paths: {}
components: 
  schemas:
    dataValues:
      type: object
      additionalProperties: false
      properties:
        service_type:
          type: string
          default: ClusterIP
          example: NodePort 
```

---

###  🟡  Scenario 3: Multiple arguments to `@schema/examples` creates a single `example` key

**Given** the following schema exists
```yaml
#! schema.yml
#@data/values-schema
---
#@schema/examples ("Exposes the Service on each Node's IP at a static port", "NodePort"), ("Loadbalancer",), ("Maps the Service to the contents of the externalName field", "ExternalName")
service_type: ClusterIP
```
**When** I execute `ytt -f schema.yml --data-values-schema-inspect -o openapi-v3`
**Then** I should see only the first example
```yaml
openapi: 3.0.0
info:
  version: 0.1.0
  title: Openapi schema generated from ytt Data Values Schema
paths: {}
components: 
  schemas:
    dataValues:
      type: object
      additionalProperties: false
      properties:
        service_type:
          type: string
          default: ClusterIP
          x-example-description: Exposes the Service on each Node's IP at a static port
          example: NodePort  
```
---

### :red_circle:  Scenario 4: Error when `@schema/examples` are a different type than the `type` key

The Spec describes this key as:
> A free-form property to include an example of an instance for this schema. To represent examples that cannot be naturally represented in JSON or YAML, a **string** value can be used to contain the example with escaping where necessary.

**Given** the following schema exists
```yaml
#! schema.yml
#@data/values-schema
---
#@schema/examples ("Zero value", 0)
enabled: false
```
**When** I execute `ytt -f schema.yml --data-values-schema-inspect -o openapi-v3`
**Then** I see an error that the value in the array must be EITHER the type of the key, OR a string
```
Invalid schema
==============

syntax error in @schema/examples annotation
schema.yml:
    |
  4 | enabled: false
    |

    = found: integer (in @schema/examples above this item)
    = expected: tuple with value of boolean or string
```

Link to the `@schema/examples` [Schema hackmd](https://hackmd.io/pODV3wzbT56MbQTxbQOOKQ?view#schemaexamples0

**Comment by cari-lynn** [comment]
This story needs:
* ✅ to determine whether it should be `@schema/examples` or `@schema/example`
* a more complex type scenario 
Relevant [slack thread](https://kubernetes.slack.com/archives/CH8KCCKA5/p1638820969129600)

**Comment by gcheadle-vmware** [comment]
Released in ytt v0.39.0
